src/gt4py/stencil_object.py

```diff
@@ -132,11 +137,13 @@ def _get_max_domain(self, field_args, origin):
         -------
             `Shape`: the maximum domain size.
         """
-        max_domain = Shape([np.iinfo(np.uintc).max] * self.domain_info.ndims)
-        shapes = {name: Shape(field.shape) for name, field in field_args.items()}
-        for name, shape in shapes.items():
-            upper_boundary = Index(self.field_info[name].boundary.upper_indices)
-            max_domain &= shape - (Index(origin[name]) + upper_boundary)
+        large_val = np.iinfo(np.uintc).max
+        max_domain = Shape([large_val] * self.domain_info.ndims)
+        for name, field in field_args.items():
+            field_mask = self._get_field_mask(field)
```

Comment:
It is probably better to get the mask based on the axes in the field info, rather than the passed field, since then it is based on what is required rather than something that may have been provided wrongly.

Reply:
Excellent idea. Implemented in 75f055a.